addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/ContentSecurityPolicyMissingScannerUnitTest.java

```diff
@@ -0,0 +1,166 @@
+    protected ContentSecurityPolicyMissingScanner createScanner() {
+        return new ContentSecurityPolicyMissingScanner();
+    }
+
+    @Test
+    public void givenMissingCspHeaderThenAlertRaised() throws Exception {
+        // Given
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_HTML});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertContentSecurityPolicyAlertRaised();
+    }
+
+    @Test
+    public void givenOnlyCspHeaderAtMediumAlertThresholdThenNoAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.MEDIUM);
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_HTML, HEADER_CSP});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void givenOnlyCspHeaderAtLowAlertThresholdThenAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.LOW);
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_HTML, HEADER_CSP});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertContentSecurityPolicyAlertRaised();
+    }
+
+    @Test
+    public void givenAllCspHeadersAtLowAlertThresholdThenNoAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.LOW);
+        String[] headers = {HEADER_HTML, HEADER_CSP, HEADER_X_CSP, HEADER_WEBKIT_CSP};
+        HttpMessage msg = createHttpMessageWithHeaders(headers);
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void givenTextContentTypeAtMediumAlertThresholdThenNoAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.MEDIUM);
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_TEXT});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void givenTextContentTypeAtLowAlertThresholdThenAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.LOW);
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_TEXT});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertContentSecurityPolicyAlertRaised();
+    }
+
+    @Test
+    public void givenReportOnlyCspThenInfoAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.MEDIUM);
+        String[] headers = {HEADER_HTML, HEADER_CSP, HEADER_REPORT_ONLY};
```

Comment:
Should also test without `HEADER_CSP` to assert that it raises the two alerts.

Reply:
I re-read this test and it doesn't make sense, not sure the situation where you'd have both headers set 🤔 .  I'm thinking the scanner should give a different alert to indicate that browsers are going to be confused.  What do you think?

Reply:
That's actually allowed, a use case: https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP#Testing_your_policy

Reply:
Thanks for that.  There there are tests for

-  Report Only
-  Report Only with CSP